src/Context/ForkContext.php

```diff
@@ -0,0 +1,170 @@
+
+use Amp\ByteStream\StreamChannel;
+use Amp\Cancellation;
+use Amp\Parallel\Context\Internal\AbstractContext;
+use Amp\Parallel\Ipc\IpcHub;
+use Amp\Serialization\NativeSerializer;
+use Amp\Serialization\Serializer;
+use Amp\TimeoutCancellation;
+use Revolt\EventLoop;
+use Revolt\EventLoop\Driver\UvDriver;
+
+/**
+ * USE AT YOUR OWN RISK! This context is not used by default in {@see DefaultContextFactory} because the timing of its
+ * creation must be purposeful and situational.
+ *
+ * Forking is not recommended at arbitrary points in an application since the entire state of the parent process is
+ * inherited into the child process, including the event-loop!
+ *
+ * Forking very early in an application, may be beneficial to copy state from the parent that was expensive to set up.
+ *
+ * This context is NOT compatible with event-loop extensions such as ext-uv or ext-ev.
+ *
+ * @template-covariant TResult
+ * @template-covariant TReceive
+ * @template TSend
+ * @extends AbstractContext<TResult, TReceive, TSend>
+ */
+final class ForkContext extends AbstractContext
+{
+    private const DEFAULT_START_TIMEOUT = 5;
+
+    public static function isSupported(): bool
+    {
+        return \function_exists('pcntl_fork')
+            && !EventLoop::getDriver() instanceof UvDriver;
+    }
+
+    /**
+     * @param string|non-empty-list<string> $script Path to PHP script or array with first element as path and
+     *     following elements options to the PHP script (e.g.: ['bin/worker.php', 'Option1Value', 'Option2Value']).
+     * @param positive-int $childConnectTimeout Number of seconds the child will attempt to connect to the parent
+     *      before failing.
+     *
+     * @throws ContextException If starting the process fails.
+     */
+    public static function start(
+        IpcHub $ipcHub,
+        string|array $script,
+        ?Cancellation $cancellation = null,
+        int $childConnectTimeout = self::DEFAULT_START_TIMEOUT,
+        Serializer $serializer = new NativeSerializer(),
+    ): self {
+        $key = $ipcHub->generateKey();
+
+        // Fork
+        if (($pid = \pcntl_fork()) < 0) {
+            throw new ContextException("Forking failed: " . \posix_strerror(\posix_get_last_error()));
+        }
+
+        // Parent
+        if ($pid > 0) {
+            try {
+                $socket = $ipcHub->accept($key, $cancellation);
+                $ipcChannel = new StreamChannel($socket, $socket, $serializer);
+
+                $socket = $ipcHub->accept($key, $cancellation);
+                $resultChannel = new StreamChannel($socket, $socket, $serializer);
+            } catch (\Throwable $exception) {
+                $cancellation?->throwIfRequested();
+
+                throw new ContextException("Connecting failed after forking", previous: $exception);
+            }
+
+            return new self($pid, $ipcChannel, $resultChannel);
+        }
+
+        // Child
+        \define("AMP_CONTEXT", "parallel");
```

Comment:
Is this the correct value or due to copy and paste?

Reply:
Oops, yes, forgot this constant isn't the library name, but the context name.